File: 백준/구현/BOJ_1010.py
# 백준 1010 다리 놓기
# 21.01.17

t = int(input())
ans = []

combi = [[0] * 31 for _ in range(31)]
combi[1][0] = 1
combi[1][1] = 1
for i in range(2, 31):
    combi[i][0] = 1
    for j in range(1, 31):
        combi[i][j] = combi[i-1][j-1] + combi[i-1][j]

while t:
    n, m = map(int, input().split())
    ans.append(combi[m][n])

    t -= 1

for i in range(len(ans)):
    print(ans[i])

File: boj/Combinatorics/BOJ_1010.swift
// 백준 1010 다리 놓기
// 21.05.29

import Foundation

var dp = [[Int]](repeating: [Int](repeating: 1, count: 30), count: 30)

for i in 2..<30 {
  for j in 1..<i {
    // nCr = n-1 C r + n-1 C r-1
    dp[i][j] = dp[i-1][j-1] + dp[i-1][j]
  }
}

let t = Int(readLine()!)!

for _ in 0..<t {
  let input = readLine()!.split(separator: " " ).map { Int(String($0))! }
  print(dp[input[1]][input[0]])
}